### PR TITLE
[Maintenance] Converting dateRange and useReport to tsx

### PR DESCRIPTION
**Pull request by ameekSinghUniversityAcc**
Link to issue: https://github.com/actualbudget/actual/issues/1483

### Replaced DateRange.js with DateRange.tsx
- Added types to function props
- Added types to function return value
- Replaced the direct modification of the props with constants (`parseStart` and `parseEnd`)
- Replaced deprecated Date fn `getYear()` with `getFullYear()` (see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getYear)
- Formatted new file with linter

### Replaced useReport.js with useReport.tsx
- Added prop types

### CODE REVIEW

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 import { theme } from '../../style';
 import Block from '../common/Block';
 
-function DateRange({ start, end }) {
-  start = d.parseISO(start);
-  end = d.parseISO(end);
+type DateRangeProps = {
+  start: string | Date;
+  end: string | Date;
+};
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:11]: **💬 suggestion:** ‏the input will always be a `string`. But I see why you did this (to re-use `start` & `end` variables). Lets solve that in a different way (will comment below).
```suggestion
type DateRangeProps = {
  start: string;
  end: string;
};
```

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 
-  let content;
-  if (start.getYear() !== end.getYear()) {
+function DateRange({ start, end }: DateRangeProps): ReactElement {
+  start = d.parseISO(start as string);
+  end = d.parseISO(end as string);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:15]: **💬 suggestion:** ‏here's how we can solve it

```suggestion
function DateRange({ start: startProp, end: endProp }: DateRangeProps): ReactElement {
  const start = d.parseISO(startProp);
  const end = d.parseISO(endProp);
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Thanks for doing this! Always nice to see TypeScript migration PRs.

And welcome to the Actual community. :)

File: packages/desktop-client/src/components/reports/DateRange.tsx (hunk 2 above)
Comment (ameekSinghUniversityAcc) [reply packages/desktop-client/src/components/reports/DateRange.tsx]: Ahhh, that's a good call! Thank you so much!! 🙏
Pushed a fix to the PR!

**Review by MatissJanis — APPROVED** [review]
Perfect

**Pull request merged**